File: Easy/1_TwoSum.php
<?php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $i = 0;
        while($i < (count($nums))) {
            $a = $target - $nums[$i];
            $b = array_search($a, array_slice($nums, ($i + 1)));
            if (gettype($b) == 'integer') {
                return [$i, ($b + $i + 1)];
            }
            $i++;
        }        
        return [];
    }
}

File: Easy/1_TwoSum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    seen = {}

    nums.each_with_index do |num, i|
        complement = target - num
        if seen.key?(complement)
            return [seen[complement], i]
        end
        seen[num] = i
    end
end